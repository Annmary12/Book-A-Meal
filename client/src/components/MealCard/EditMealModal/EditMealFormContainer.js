import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import MealForm from '../../Meals/MealForm/MealForm.js';
import { updateMeal } from '../../../actions/meals.js';
import mealInfoSchema from '../../../utils/validation-schemas/mealInfoSchema.js';

export const editMealFormConfig = {
  mapPropsToValues: ({ meal }) => meal,
  validationSchema: mealInfoSchema,
  handleSubmit: async (values, { setSubmitting, props }) => {
    setSubmitting(true);

    await props.dispatch(updateMeal(values, setSubmitting));
    props.handleClose();

    setSubmitting(false);
  }
};

export default compose(
  connect(),
  withFormik(editMealFormConfig)
)(MealForm);
