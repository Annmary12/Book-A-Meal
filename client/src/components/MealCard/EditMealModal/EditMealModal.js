import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import EditMealFormContainer from './EditMealFormContainer.js';

/**
 * @class AddMealModal
 */
const EditMealModal = ({ isOpen, handleClose, meal }) => (
  <ReactModal
    isOpen={isOpen}
    onRequestClose={handleClose}
    overlayClassName="react-modal-overlay"
    className="react-modal-body"
    closeTimeoutMS={150}
  >
    <EditMealFormContainer
      handleClose={handleClose}
      meal={meal}
    />
  </ReactModal>
);

EditMealModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  meal: PropTypes.shape({
    id: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired
  }).isRequired
};

export default EditMealModal;
