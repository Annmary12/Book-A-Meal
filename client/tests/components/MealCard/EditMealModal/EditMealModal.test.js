import React from 'react';
import EditMealModal from '../../../../src/components/MealCard/EditMealModal/EditMealModal';

describe('EditMealModal component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <EditMealModal />
    );

    expect(wrapper).toMatchSnapshot();
  });
});