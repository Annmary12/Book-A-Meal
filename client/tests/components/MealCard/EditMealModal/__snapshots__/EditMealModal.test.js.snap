// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMealModal component should match snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="react-modal-body"
  closeTimeoutMS={150}
  isOpen={false}
  overlayClassName="react-modal-overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <Connect(C) />
</Modal>
`;
